### PR TITLE
Revive parity_setMinGasPrice RPC call

### PR DESCRIPTION
**Pull request by HCastano**
This RPC call was deprecated a while back during a refactoring, but there's been a feature request (#9125) wanting it to be restored.

This PR gives the people what they want.

### CODE REVIEW

File: ethcore/src/miner/miner.rs [hunk 1]
```diff
@@ -865,6 +865,26 @@ impl miner::MinerService for Miner {
 		self.params.read().gas_range_target.0 / 5
 	}
 
+	fn set_minimal_gas_price(&self, new_price: U256) -> Result<bool, ()> {
+		if cfg!(feature = "price-info") {
+			warn!("Can't update fixed gas price while automatic gas calibration is enabled.");
```
Comment (tomusdrw) [inline ethcore/src/miner/miner.rs]: That's not 100% true, the fact that there is an option to set non-fixed `GasPricer` doesn't imply that it's actually set.
I think this `if` should rather be replaced with something like:
```
if let GasPricer::Fixed(mut val) = *self.gas_pricer.lock() {
    // update and recalibrate
   *val = new_price;
   // ...
} else {
  // Stringified error would be returned to RPC caller which imho would be nice.
  return Err("...");
}
```

File: ethcore/src/miner/miner.rs [hunk 2]
```diff
@@ -865,6 +865,26 @@ impl miner::MinerService for Miner {
...
+			return Err(());
+		}
+
+		trace!(target: "miner", "minimal_gas_price: recalibrating fixed...");
+		*self.gas_pricer.lock() = GasPricer::new_fixed(new_price);
+
+		let txq = self.transaction_queue.clone();
+		let mut options = self.options.pool_verification_options.clone();
+		self.gas_pricer.lock().recalibrate(move |gas_price| {
```
Comment (tomusdrw) [inline ethcore/src/miner/miner.rs]: Is it worth keeping it DRY? and re-using the same method as for per-block update?

File: ethcore/src/miner/miner.rs [hunk 3]
```diff
@@ -1646,4 +1666,19 @@ mod tests {
 
 		assert!(miner.is_currently_sealing());
 	}
+
+	#[test]
+	#[cfg(not(feature = "price-info"))]
```
Comment (tomusdrw) [inline ethcore/src/miner/miner.rs]: Would be good to have test for both cases.

File: rpc/src/v1/impls/parity_set.rs [hunk 4]
```diff
@@ -69,9 +69,11 @@ impl<C, M, U, F> ParitySet for ParitySetClient<C, M, U, F> where
 	F: Fetch + 'static,
 {
 
-	fn set_min_gas_price(&self, _gas_price: U256) -> Result<bool> {
-		warn!("setMinGasPrice is deprecated. Ignoring request.");
-		Ok(false)
+	fn set_min_gas_price(&self, gas_price: U256) -> Result<bool> {
+		match self.miner.set_minimal_gas_price(gas_price.into()) {
+			Ok(success) => Ok(success),
+			Err(_) => Err(errors::request_rejected()),
```
Comment (tomusdrw) [inline rpc/src/v1/impls/parity_set.rs]: I think this error is reserved for signer, maybe let's return `unsupported` with explanation here?

File: rpc/src/v1/tests/mocked/parity_set.rs [hunk 5]
```diff
@@ -106,7 +106,24 @@ fn rpc_parity_set_min_gas_price() {
...
 
 	let request = r#"{"jsonrpc": "2.0", "method": "parity_setMinGasPrice", "params":["0xcd1722f3947def4cf144679da39c4c32bdc35681"], "id": 1}"#;
-	let response = r#"{"jsonrpc":"2.0","result":false,"id":1}"#;
+	let response = r#"{"jsonrpc":"2.0","result":true,"id":1}"#;
+
+	assert_eq!(io.handle_request_sync(request), Some(response.to_owned()));
+}
+
+#[test]
+#[cfg(feature = "price-info")]
```
Comment (tomusdrw) [inline rpc/src/v1/tests/mocked/parity_set.rs]: I think `rpc` doesn't' have this feature so the test should be enable by default, but `TestMiner` might need a way to mock the error response.

File: ethcore/src/miner/miner.rs (hunk 3 above)
Comment (tomusdrw) [reply ethcore/src/miner/miner.rs]: I think the features setting is irrelevant for this test, let's just remove this line.

File: rpc/src/v1/tests/mocked/parity_set.rs (hunk 5 above)
Comment (tomusdrw) [reply rpc/src/v1/tests/mocked/parity_set.rs]: Still valid?

Comment (HCastano) [reply rpc/src/v1/tests/mocked/parity_set.rs]: I've removed the need for the `cfg` flag in [8f04c73](/paritytech/parity-ethereum/pull/10294/commits/8f04c73199f1314c952e6ef24c5af986973f14c2).

File: ethcore/src/miner/miner.rs (hunk 3 above)
Comment (HCastano) [reply ethcore/src/miner/miner.rs]: I think it was left over from earlier, but let me double check.

File: ethcore/src/miner/miner.rs (hunk 2 above)
Comment (HCastano) [reply ethcore/src/miner/miner.rs]: We only call `recalibrate()` twice, so for now I'm going to say no unless somebody else has a strong opinion about it.

**Comment by tomusdrw** [comment]
@HCastano seems like the merge went wrong, can you check?

**Pull request merged**